### PR TITLE
Export - small cleanup

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Minor tidy up on export

Before
----------------------------------------
parameters $saveFile  & $batchItems are passed in but never set in calling functions. Export mode passed in

After
----------------------------------------
unneeded params not passed in. processor object passed in instead of exportMode. This is part of a bigger picture effort to move processing off to that object rather than passing around parameters

Technical Details
----------------------------------------
getExportFileName appears to be called from 2 places. One is on the file where it always passed no params & gets the same result so this is now hard-coded. The second is moved to the ExportProcessor class, with some handling for the possibility that mode might be invalid - but I think that is never true.

Comments
----------------------------------------
Follow up on https://github.com/civicrm/civicrm-core/pull/12586

### CODE REVIEW

File: CRM/Export/BAO/Export.php [hunk 1]
```diff
@@ -749,7 +710,7 @@ public static function exportCustom($customSearchClass, $formValues, $order) {
       $rows[] = $row;
     }
 
-    CRM_Core_Report_Excel::writeCSVFile(self::getExportFileName(), $header, $rows);
+    CRM_Core_Report_Excel::writeCSVFile(ts('CiviCRM Contact Search'), $header, $rows);
     CRM_Utils_System::civiExit();
```
Comment (mattwire) [inline CRM/Export/BAO/Export.php]: Not sure about this change as it duplicates a string - do we need to make this change?

**Comment by mattwire** [comment]
I've looked through this and made one trivial comment.  The code looks good to me, it's cleanup and will make the code easier to maintain.  @twomice are you able to run this?

File: CRM/Export/BAO/Export.php (hunk 1 above)
Comment (eileenmcnaughton) [reply CRM/Export/BAO/Export.php]: The point is to stop it calling what is essentially an unrelated function for a different export process - ie. to break dumb interdependencies

Comment (mattwire) [reply CRM/Export/BAO/Export.php]: Right. Totally makes sense then :-)

**Comment by eileenmcnaughton** [comment]
test this please

**Comment by eileenmcnaughton** [comment]
what a beautiful PR review :-) - merging

**Pull request merged**